community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -138,6 +137,64 @@ final CountsOperations counting()
         return operations.counting();
     }
 
+    // <DataReadCursors>
+
+    @Override
+    public NodeCursor nodeCursor( long nodeId )
+    {
+        return statement.acquireNodeCursor().init(statement.getStoreStatement().acquireSingleNodeCursor().init( nodeId ));
+    }
+
+    @Override
+    public RelationshipCursor relationshipCursor( long relId )
+    {
+        return statement.acquireRelationshipCursor().init( statement.getStoreStatement().acquireSingleRelationshipCursor().init( relId ) );
+    }
+
+    @Override
+    public RelationshipCursor relationshipCursorGetAll(  )
+    {
+        return statement.acquireRelationshipCursor().init( dataRead().relationshipCursorGetAll(statement) );
+    }
+
+    @Override
+    public NodeCursor nodeCursorGetForLabel( int labelId )
+    {
+        return statement.acquireNodeCursor().init( statement.getStoreStatement().acquireIteratorNodeCursor().init(
+                nodesGetForLabel( labelId ) ));
+    }
+
+    @Override
+    public NodeCursor nodeCursorGetFromIndexLookup(  IndexDescriptor index,
+            Object value ) throws IndexNotFoundKernelException
+    {
+        return statement.acquireNodeCursor().init( statement.getStoreStatement().acquireIteratorNodeCursor().init( nodesGetFromIndexLookup(
+                index, value )));
+    }
+
+    @Override
+    public NodeCursor nodeGetFromIndexScan( IndexDescriptor index ) throws IndexNotFoundKernelException
+    {
+        return statement.acquireNodeCursor().init( statement.getStoreStatement().acquireIteratorNodeCursor().init( nodesGetFromIndexScan(
+                index )));
+    }
+
+    @Override
+    public NodeCursor nodeCursorGetUniqueFromIndexLookup( IndexDescriptor index,
+            Object value ) throws IndexNotFoundKernelException, IndexBrokenKernelException
+    {
+        return statement.acquireNodeCursor().init( statement.getStoreStatement().acquireIteratorNodeCursor().init( PrimitiveLongCollections.singleton(
+                        nodeGetUniqueFromIndexLookup( index, value ) )));
+    }
+
+    @Override
+    public NodeCursor nodeCursorGetAll( )
+    {
+        statement.assertOpen();
```

Comment:
should we assert that the statement is open also on the other cursor related method above?